share/wake/lib/system/io.wake

```diff
@@ -39,6 +39,35 @@ target writeImp inputs mode path content =
 
         makeRunner "write" (\_ Pass 0.0) pre post virtualRunner
 
+    # There are a couple likely bad paths that we don't want the user writing to
+    # so we give good error messages for these cases
+    require False = path ==* ""
+    else
+        failWithError
+        "Attempt to write to the path of the empty string. This was likely a mixup between the path and the content."
+
+    require False = path ==* "."
+    else failWithError "Attempt to write to write to the root workspace"
+
+    # We don't want `write` to write to anything outside of the workspace via
+    # a relative path
+    require False = matches `\.\..*` path
+    else failWithError "Attempt to write outside of the workspace"
```

Comment:
This probably needs to be run through a relativizer of some sort (I don't think it is yet, but I might have missed something) to catch `subdir/../..`.  It's also filtering out things like `..inRoot` which are most definitely *bad* names but which are still in the workspace.

Reply:
All paths leading into this function simplify the path before passing it to the implementation

Reply:
Great, thanks!